attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java

```diff
@@ -92,17 +97,28 @@ public void destroy(@Observes @Destroyed(ApplicationScoped.class) final Object c
 
     @Override
     public SubscriberBuilder<? extends Message<?>, Void> getSubscriberBuilder(final Config config) {
-        final PubSubConfig pubSubConfig = new PubSubConfig(projectId, getTopic(config), getCredentialPath(config),
+        final PubSubConfig pubSubConfig = new PubSubConfig(getProjectId(config), getTopic(config), getCredentialPath(config),
                 mockPubSubTopics, host.orElse(null), port.orElse(null));
 
         return ReactiveStreams.<Message<?>> builder()
                 .flatMapCompletionStage(message -> CompletableFuture.supplyAsync(() -> {
-                    createTopic(pubSubConfig);
+                    if (isUseAdminClient(config)) {
+                        createTopic(pubSubConfig);
+                    }
                     return await(pubSubManager.publisher(pubSubConfig).publish(buildMessage(message)));
                 }, executorService))
                 .ignore();
     }
 
+    private String getProjectId(Config config) {
+        return config.getOptionalValue("project-id", String.class)
+                .orElse(projectId);
+    }
+
+    boolean isUseAdminClient(Config config) {
+        return config.getOptionalValue("use-admin-client", Boolean.class).orElse(true);
```

Comment:
At some point, we really need to switch to @ConnectorAttribute and avoid that kind of parsing.

Reply:
I can do that in future. You can create a task / issue and assign to me when u want.